lib/puppet/util/windows/registry.rb

```diff
@@ -31,16 +31,24 @@ def open(name, path, mode = KEY_READ | KEY64, &block)
     end
 
     def values(subkey)
+      orig_enc = Encoding.default_internal
+      Encoding.default_internal = Encoding::UTF_8
```

Comment:
When it converts to the native locale_charmap (in my case that is `IBM437`), instead of a nice successful conversion, you get this nice fun instead:

```
Encoding::UndefinedConversionError:
       U+2013 to IBM437 in conversion from UTF-16LE to UTF-8 to IBM437
```

`Encoding.default_internal` is `nil` by default. Falling back to rely on whatever the native encoding is doesn't handle converting strings correctly. It's probably better to go with whatever the registry itself adheres to. And stay there. 

Sounds like UTF-16LE?